src/app/content/studyGuides/components/ShowStudyGuides.tsx

```diff
@@ -44,26 +44,30 @@ const ShowStudyGuides = () => {
     setShowGoToTop(false);
   };
 
-  const fetchMoreHighlights = React.useCallback(() => {
-    if (isLoading || !ref.current) { return; }
-    const scrollBottom = ref.current.scrollHeight - ref.current.offsetHeight - ref.current.scrollTop;
-    if (scrollBottom <= loadMoreDistanceFromBottom && hasMoreResults) {
-      dispatch(loadMoreStudyGuides());
+  const updateBackToTop = (refElement: HTMLElement) => {
+    setShowGoToTop(refElement.scrollTop > 0);
+  };
+
+  const fetchMoreStudyGuides = (refElement: HTMLElement) => {
+    if (isLoading || !hasMoreResults) {
+      return;
     }
-  }, [ref, dispatch, isLoading, hasMoreResults]);
 
-  React.useEffect(() => {
-    const refElement = ref.current;
-    if (refElement) {
-      setShowGoToTop(refElement.scrollTop > 0);
-      refElement.addEventListener('scroll', fetchMoreHighlights);
+    const scrollBottom = refElement.scrollHeight - refElement.offsetHeight - refElement.scrollTop;
+    if (scrollBottom <= loadMoreDistanceFromBottom) {
+      dispatch(loadMoreStudyGuides());
     }
-    return () => refElement ? refElement.removeEventListener('scroll', fetchMoreHighlights) : undefined;
-  }, [fetchMoreHighlights]);
+  };
 
   return (
     <StudyGuidesBody
       ref={ref}
+      onScroll={() => {
+        const refElement = assertNotNull(ref.current, 'Scroll event on a non existing DOM node');
```

Comment:
Can this actually happen? From what I've read refs are set when component mounts and set back to null when it unmounts. 

Also, was there a reason why the event handler was added to the node instead of using React's `onScroll`?